bundles/io/org.eclipse.smarthome.io.net/src/main/java/org/eclipse/smarthome/io/net/http/HttpRequestBuilder.java

```diff
@@ -0,0 +1,127 @@
+ */
+public class HttpRequestBuilder {
+    private static final int DEFAULT_TIMEOUT_IN_MS = 5_000;
+
+    private final String method;
+    private final String url;
+    private InputStream inputStream;
+    private int timeoutMilliseconds = DEFAULT_TIMEOUT_IN_MS;
+    private Properties headers;
+    private String contentType;
+
+    /**
+     * Private constructor, to hide away the construction behind some factory methods
+     *
+     * @param method Http method to be used
+     * @param url url to fetch or post data to
+     */
+    private HttpRequestBuilder(String method, String url) {
+        this.method = method;
+        this.url = url;
+    }
+
+    /**
+     * Construct an http request builder to get data from an url
+     *
+     * @param url to fetch the data from
+     * @return a request builder to construct and complete the request
+     */
+    public static HttpRequestBuilder getFrom(String url) {
+        return new HttpRequestBuilder(HttpUtil.METHOD_GET, url);
+    }
+
+    /**
+     * Construct an http request builder to post data to a url
+     *
+     * @param url to post data to
+     * @return a request builder to construct and complete the request
+     */
+    public static HttpRequestBuilder postTo(String url) {
+        return new HttpRequestBuilder(HttpUtil.METHOD_POST, url);
+    }
+
+    /**
+     * Add content to this request
+     *
+     * @param content a string containing the data to be pushed to the url
+     * @return a request builder to construct and complete the request
+     */
+    public HttpRequestBuilder withContent(String content) {
+        this.inputStream = new ByteArrayInputStream(content.getBytes(StandardCharsets.UTF_8));
+        return this;
+    }
+
+    /**
+     * Add an additional header to the request
+     *
+     * @param header name of the header, eg. Content-Type
+     * @param value value of the header, eg. "application/json"
+     * @return a request builder to construct and complete the request
+     */
+    public HttpRequestBuilder withHeader(String header, String value) {
+        if (headers == null) {
+            headers = new Properties();
+        }
+
+        headers.put(header, value);
+        return this;
+    }
+
+    /**
+     * Add a content type for this request
+     *
+     * @param contentType a string containing the data to be pushed to the url
+     * @return a request builder to construct and complete the request
+     */
+    public HttpRequestBuilder withContentType(String contentType) {
+        this.contentType = contentType;
+        return this;
+    }
+
+    /**
+     * Add a timeout for this request
+     *
+     * @param amount pas the value of timeout in the unit which is applied as the second parameter
+     * @param unit a TimeUnit eg. TimeUnit.SECONDS
+     * @return a request builder to construct and complete the request
+     */
+    public HttpRequestBuilder withTimeout(int amount, TimeUnit unit) {
```

Comment:
Instead of amount and time unit, we could provide a method with `Duration`.

```java
public HttpRequestBuilder withTimeout(Duration duration) {
...
}
```